packages/reports/addon/components/report-builder.js

```diff
@@ -39,5 +40,55 @@ export default Component.extend({
   allTables: computed(function() {
     let metadataService = get(this, 'metadataService');
     return metadataService.all('table').sortBy('longName');
-  })
+  }),
+
+  actions: {
+    /**
+     * @action expandFilters
+     * @param {Function} shouldExpand
+     */
+    expandFilters(shouldExpand) {
+      const { isFiltersCollapsed, onUpdateFiltersCollapsed } = this;
+
+      if (isFiltersCollapsed && typeof onUpdateFiltersCollapsed === 'function' && shouldExpand()) {
+        onUpdateFiltersCollapsed(false);
+      }
+    },
+
+    /**
+     * @action onToggleDimFilter
+     * @param {Object} dimension
+     */
+    onToggleDimFilter(dimension) {
+      this.send('expandFilters', () => arr(this.request.filters).findBy('dimension', dimension));
+    },
+
+    /**
+     * @action onToggleMetricFilter
+     * @param {Object} metric
+     */
+    onToggleMetricFilter(metric) {
+      this.send('expandFilters', () =>
+        arr(this.request.having).find(having => get(having, 'metric.metric.name') === get(metric, 'name'))
```

Comment:
We can remove the `arr` wrapper or change this to use `findBy`

`arr(this.request.having).findBy('metric.metric.name', get(metric, 'name'))`